clients/fides-js/src/fides-tcf.ts

```diff
@@ -263,35 +236,32 @@ const init = async (config: FidesConfig) => {
     ...getInitialCookie(config),
     ...overrides.consentPrefsOverrides?.consent,
   };
-  if (config.options.fidesString) {
-    const { cookie: updatedCookie, success } = updateFidesCookieFromString(
-      cookie,
-      config.options.fidesString,
-      config.options.debug,
-      overrides.consentPrefsOverrides?.version_hash
-    );
-    if (success) {
+  // Update the fidesString if we have an override and the TC portion is valid
+  const { fidesString } = config.options;
+  if (fidesString) {
+    try {
+      // Make sure TC string is valid before we assign it
+      const { tc: tcString } = decodeFidesString(fidesString);
+      TCString.decode(tcString);
+      const updatedCookie: Partial<FidesCookie> = {
+        fides_string: fidesString,
+        tcf_version_hash:
+          overrides.consentPrefsOverrides?.version_hash ??
+          cookie.tcf_version_hash,
+      };
       Object.assign(cookie, updatedCookie);
+    } catch (error) {
+      debugLog(
+        config.options.debug,
+        `Could not decode tcString from ${fidesString}, it may be invalid. ${error}`
+      );
     }
```

Comment:
It seems like this got... a lot simpler! That's actually a pretty good sign 👍

Reply:
yes, I agree! was a pleasant surprise 😄